### PR TITLE
Move `Trivia` from gyb to code gen

### PR DESCRIPTION
**Pull request by kimdv**
There is an error when generating `Swift/Character.swift:177: Fatal error: Can't form a Character from a String containing more than one extended grapheme cluster`

### CODE REVIEW

**Comment by ahoppen** [comment]
> There is an error when generating `Swift/Character.swift:177: Fatal error: Can't form a Character from a String containing more than one extended grapheme cluster`

in SyntaxSupport/gyb_generated/Trivia.swift, you have `Character(#"\t"#)`. Because of the `#` it’s a raw string literal and thus `\t` is interpreted as two characters, the backlash and `t`. Thus you can’t form a character from it. If you change it to `Character("\t")`, you get the literal tab character, which works.

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TriviaFile.swift [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+  }
+
+  ExtensionDeclSyntax("extension RawTriviaPiece") {
+    VariableDeclSyntax(
+      modifiers: [DeclModifierSyntax(name: .keyword(.public))],
+      name: IdentifierPatternSyntax("byteLength"),
+      type: TypeAnnotationSyntax(type: TypeSyntax("Int"))) {
+        SwitchStmtSyntax(expression: ExprSyntax("self")) {
+          for trivia in TRIVIAS {
+            if trivia.isNewLine {
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TriviaFile.swift:385]: Same suggestion as above that we can make this not rely on `isNewLine`

**Review by ahoppen — COMMENTED** [review]
Overall LGTM, just a few code style comments.

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TriviaFile.swift [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+  ExtensionDeclSyntax(#"""
+    extension Trivia: CustomDebugStringConvertible {
+      public var debugDescription: String {
+        if count == 1, let first = first {
+          return first.debugDescription
+        }
+        return "[" + map(\.debugDescription).joined(separator: ", ") + "]"
+      }
+    }
+    """#)
```
[the remark this answers is not in the document]
Comment (kimdv) [reply CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TriviaFile.swift:209]: There is `"` inside the string L206. I think it makes it easier to read then

**Comment by kimdv** [comment]
@swift-ci please test

File: Sources/SwiftSyntax/generated/Trivia.swift [hunk 3]
```diff
@@ -146,165 +162,171 @@ extension TriviaPiece {
...
-  /// Returns a piece of trivia for some number of '\n' characters.
+  
+  /// Returns a piece of trivia for some number of "\n" characters.
   public static func newlines(_ count: Int) -> Trivia {
     return [.newlines(count)]
   }
-
-  /// Gets a piece of trivia for '\n' characters.
+  
+  /// Gets a piece of trivia for '"\n"' characters.
```
Comment (ahoppen) [inline Sources/SwiftSyntax/generated/Trivia.swift:244]: This contains two quotes: `'` and `"`.

**Review by ahoppen — APPROVED** [review]
LGTM

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
Depends on #1258

**Comment by kimdv** [comment]
@swift-ci please test

**Pull request merged**